packages/api/src/tasks/checkAlerts/providers/default.ts

```diff
@@ -198,7 +199,7 @@ async function loadAlert(
   if (!v) {
     throw new Error(`provider did not set key ${conn.id} before appending`);
   }
-  v.alerts.push({ ...details, previous });
+  v.alerts.push({ ...details, previous, previousMap: previousAlerts });
```

Comment:
Do we need to maintain both the `previous` and `previousMap` fields here, or can we just pass `previousMap`?

I'm also wondering - in `processAlert` we use `previous` to get the `dateRange` and for the `Skipped to check alert since the time diff is still less than 1 window size` check. For grouped alerts, won't previous always be undefined? Or are we saving an alert history without a group key even for alerts with groups?